src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -356,5 +438,76 @@ public void WriteTo(TextWriter writer, HtmlEncoder encoder)
 
             _postElement?.WriteTo(writer, encoder);
         }
+
+        private void CopyAttributesTo(IHtmlContentBuilder destination)
+        {
+            StringWriter stringWriter = null;
+
+            // Perf: Avoid allocating enumerator
+            for (var i = 0; i < (_attributes?.Count ?? 0); i++)
+            {
+                var attribute = _attributes[i];
+                destination.AppendHtml(" ");
+                destination.AppendHtml(attribute.Name);
+
+                if (attribute.Minimized)
+                {
+                    continue;
+                }
+
+                destination.AppendHtml("=\"");
+                var value = attribute.Value;
+                var htmlContent = value as IHtmlContent;
```

Comment:
The answer from @NTaylorMullen is that this will generally be an `HtmlEncodedString` or string + needs encoding.

The user could do something weird here like put a view buffer inside of an attribute. I think we're we're doing here is fine, we'd be suboptimal, but not incorrect if you did that.